engine/logic/sorter_compare.go

```diff
@@ -0,0 +1,329 @@
+		case uint16:
+			return CompareInts(int64(x), int64(y)), nil
+		case uint64:
+			return CompareInts(int64(x), int64(y)), nil
+		case float32:
+			return CompareFloats(float64(x), float64(y)), nil
+		case float64:
+			return CompareFloats(float64(x), y), nil
+		}
+	case uint64:
+		// Compare uint64 with other types
+		switch y := b.(type) {
+		case uint64:
+			return CompareInts(int64(x), int64(y)), nil
+		case int:
+			return CompareInts(int64(x), int64(y)), nil
+		case int16:
+			return CompareInts(int64(x), int64(y)), nil
+		case int32:
+			return CompareInts(int64(x), int64(y)), nil
+		case int64:
+			return CompareInts(int64(x), y), nil
+		case uint16:
+			return CompareInts(int64(x), int64(y)), nil
+		case uint32:
+			return CompareInts(int64(x), int64(y)), nil
+		case float32:
+			return CompareFloats(float64(x), float64(y)), nil
+		case float64:
+			return CompareFloats(float64(x), y), nil
+		}
+	case float32:
+		// Compare float32 with other types
+		switch y := b.(type) {
+		case float32:
+			return CompareFloats(float64(x), float64(y)), nil
+		case float64:
+			return CompareFloats(float64(x), y), nil
+		case int:
+			return CompareFloats(float64(x), float64(y)), nil
+		case int16:
+			return CompareFloats(float64(x), float64(y)), nil
+		case int32:
+			return CompareFloats(float64(x), float64(y)), nil
+		case int64:
+			return CompareFloats(float64(x), float64(y)), nil
+		case uint16:
+			return CompareFloats(float64(x), float64(y)), nil
+		case uint32:
+			return CompareFloats(float64(x), float64(y)), nil
+		case uint64:
+			return CompareFloats(float64(x), float64(y)), nil
+		}
+	case float64:
+		// Compare float64 with other types
+		switch y := b.(type) {
+		case float64:
+			return CompareFloats(x, y), nil
+		case float32:
+			return CompareFloats(x, float64(y)), nil
+		case int:
+			return CompareFloats(x, float64(y)), nil
+		case int16:
+			return CompareFloats(x, float64(y)), nil
+		case int32:
+			return CompareFloats(x, float64(y)), nil
+		case int64:
+			return CompareFloats(x, float64(y)), nil
+		case uint16:
+			return CompareFloats(x, float64(y)), nil
+		case uint32:
+			return CompareFloats(x, float64(y)), nil
+		case uint64:
+			return CompareFloats(x, float64(y)), nil
+		}
+	}
+
+	// If we reach here, the types are not comparable
+	return 0, fmt.Errorf("incompatible types: %s and %s", reflect.TypeOf(a).String(), reflect.TypeOf(b).String())
+}
+
+// CompareInts is a helper function to compare two int values and return the appropriate comparison result.
+func CompareInts(x, y int64) int {
+	if x < y {
+		return -1
+	} else if x > y {
+		return 1
+	}
+	return 0
+}
+
+// CompareFloats is a helper function to compare two float values and return the appropriate comparison result.
+func CompareFloats(x, y float64) int {
+	if x < y {
+		return -1
+	} else if x > y {
+		return 1
+	}
+	return 0
+}
+
+// CompareAny compares two variables of any type.
+func CompareAny(a, b any) int {
+
+	if a == nil && b != nil {
```

Comment:
I think these types of comparisons are what is causing the error: " unexpected end of JSON input". Since nil value is probably not getting converted correctly and tripping this error, ex:

```
{"level":"error","msg":"sort error: unexpected end of JSON input on {\"Current\":{\"ID\":\"1","time":"2025-01-29T10:17:09-08:00"}
{"level":"info","msg":"Field: $f1.content[0].attachment.size Compare 1.4676006096e+10 v \u003cnil\u003e = 1\n","time":"2025-01-29T10:17:09-08:00"}
```
probably caused by outNull creating empty map. Jsonpath sorts like '.sort("$f1.content[0].attachment.size")' appear to be working, on the no null data though.